src/roiextractors/extractors/memmapextractors/numpymemampextractor.py

```diff
@@ -0,0 +1,178 @@
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from typing import Tuple, Dict
+
+import numpy as np
+from tqdm import tqdm
+
+from ...imagingextractor import ImagingExtractor
+from typing import Tuple, Dict
+
+from ...extraction_tools import (
+    PathType,
+    DtypeType,
+)
+
+
+class NumpyMemmapImagingExtractor(ImagingExtractor):
+
+    extractor_name = "NumpyMemmapImagingExtractor"
+
+    def __init__(
+        self,
+        file_path: PathType,
+        frame_shape: Tuple[int, int],
+        sampling_frequency: float,
+        dtype: DtypeType,
+        offset: int = 0,
+        image_structure_to_axis: Dict[str, int] = None,
+    ):
+        """Class for reading optical imaging data stored in a binary format with np.memmap
+
+
+        Parameters
+        ----------
+        file_path : PathType
+            the file_path where the data resides.
+        frame_shape : tuple
+            The frame shape of the image determines how each frame looks. Examples:
+            (n_channels, rows, columns), (rows, columns, n_channels), (n_channels, columns, rows), etc.
+            Note that n_channels is 1 for grayscale and 3 for color images.
+        dtype : DtypeType
+            The type of the data to be loaded (int, float, etc.)
+        offset : int, optional
+            The offset in bytes. Usually corresponds to the number of bytes occupied by the header. 0 by default.
+        sampling_frequency : float, optional
+            The sampling frequency.
+        image_structure_to_axis : dict, optional
+            A dictionary indicating what axis corresponds to what in the memmap. The default values are:
+            dict(frame_axis=0, num_channels=1, rows=2, columns=3)
+            frame_axis=0 indicates that the first axis corresponds to the frames (usually time)
+            num_channels=1 here indicates that the first axis corresponds to the  n_channels.
+            rows=2 indicates that the rows in the image are in the second axis
+            columns=3 indicates that columns is the last axis or dimension in this structure.
+
+            Notice that this should correspond with frame_shape.
+        """
+
+        self.installed = True
+        super().__init__()
+
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        self.offset = offset
+        self.dtype = dtype
+
+        # Get the structure, apply default if not available
+        self.frame_shape = frame_shape
+        image_structure_to_axis = dict() if image_structure_to_axis is None else image_structure_to_axis
+        self.image_structure_to_axis = dict(frame_axis=0, num_channels=1, rows=2, columns=3)
```

Comment:
I don't know if it's a good idea to have a default for this argument - it's so important, I'd recommend making it required. 

I'd also recommend, if you're sure you want to keep it as a simple dictionary, we should check the keys for possible typos/missing values by asserting frame+rows+col keys are all present (channels could be optional since some binary blobs may represent grayscale and hence have no extra axis for that).

Reply:
I am still on the fence about whether this dict structure is good or not. What where the other suggestions? a tuple of two tuples (the first with the values and the second with the mapping and a data class).

Reply:
Not a tuple-of-tuples, definitely. That's one of those data structures that is less Pythonic and more reminiscent of MATLAB.

What we want is definitely something dictionary-like, but with constraints (limited possibilities) and convenience (fast, informative error message for typos).

Can always start off with a `dict` + `assert` and change it later if we can think of a better way. It is, I suppose, the simplest thing to ask of a user if not the most sophisticated from a development standpoint.

Reply:
Yes, after I thought more about it and re-read your arguments in #112 I have changed my mind. Check #115 for an implementation of a data-class for handling this.